google/cloud/iam/samples/iam_samples.cc

```diff
@@ -27,6 +27,35 @@
 #include <utility>
 
 namespace {
+
+void ExampleStatusOr(std::vector<std::string> const& argv) {
+  //! [example-status-or]
+  namespace iam = ::google::cloud::iam;
+  [](std::string const& project_id) {
+    iam::IAMClient client(iam::MakeIAMConnection());
+    int count = 0;
+    // The actual type of `service_account` is
+    // google::cloud::StatusOr<google::iam::admin::v1::ServiceAccount>, but
+    // we expect it'll most often be declared with auto like this.
+    for (auto const& service_account :
```

Comment:
Consider: `auto const& sa`, longer names do not make code more readable.  There are actual studies on that:

https://www.researchgate.net/publication/318797794_Fixing_Faults_in_C_and_Java_Source_Code_Abbreviated_vs_Full-Word_Identifier_Names